### DISCUSSION TITLE
Werkzeug Höhenmodellabfrage

### DISCUSSION THREAD

**Discussion by nicole-vlbg** [e-netze/webgis-community#111] [Ideas]
Es geht hier nochmals um das Anschreiben der Höhen (Gelände u Oberfläche) direkt in der Karte beim gesetzten Punkt wie in WebGIS 4
![image](https://github.com/gis-eni/webgis-forum/assets/124671123/de785baf-c497-419d-ad76-74320d03008b)

Wir sind uns bewusst, dass das Thema schonmal diskutiert wurde, haben aber immer wieder Anfragen von Usern die diese Funktion benötigen und mit dem derzeit gebotenen so gar nicht zufrieden sind. Auch nicht mit der Lösung über das anschreiben beim Ausdruck in der Koordinatenabfrage, was ja auch nicht wirklich eine Alternative ist. 
Wir selbst fänden es eine Bereicherung für das Werkzeug im WebGIS5. Damit macht es wieder Sinn es zusätzlich zur Koordinatenabfrage anzubieten und wir hätten glückliche User.

**Comment by gis-eni** [comment]
Das Werkzeug "Höhenmodel" ist im WebGIS 5 **deprecated**.

Das Werkzeug wurde jetzt zu einem Werkzeug XYZ zusammengefasst, wo du Koordinaten und Höhen gleichzeitig abfragen kannst. Das macht sinn, weil man die Ergebnisse herunterladen kann und die Daten so auch weiter verarbeiten kann. 

Bietet man beide Werkzeuge an, ist das verwirrend, was man wofür verwenden soll. Ein Beschriftung in der Karte (Koordinaten/Höhen) ist im XYZ Werkzeug erste einmal nicht angedacht (außer im Ausdruck).

Bietet man mehrere Höhenmodelle an, wird die Koordianten/Höhen Tabelle natürlich schnell zu breit für den seitlichen Frame und alles wird schwer lesbar. Eine Lösung kommt dazu in der neuen WebGIS Version (webgis 6.x), die ihr ja schon zum Testen herunter laden könnt. Dort lässt sich die Größe, der *SideBar* beliebig anpassen und alle Höhen sind dann auch gut lesbar:

![image](https://github.com/gis-eni/webgis-forum/assets/62870568/adbf1f2d-76f9-47de-9080-055c9ae2184d)

**Comment by ue-black** [comment]
Auch hier nochmal zur Präzisierung: Das Tool Höhenmodell könnte man dazu nützen, dass man Gelände- und Oberfläche direkt in der Karte beschriftet (das Tool XYZ bleibt dabei unverändert). So wie Nicole oben schon geschrieben hat. Das Tool Höhenmodell muss man ja auch nicht einbauen wenn man nicht will...

Dabei muss man natürlich mitdenken, dass die Höhenkoten auch beim Ausdruck dargestellt werden können. Das ist ja im Moment nicht möglich.

**Comment by gis-eni** [comment]
Ja, aber mir gefällt das eben auch nicht so. Wenn dann gehört das ins XYZ Tool rein. Dafür zwei Tools zu haben, halte ich nicht für sinnvoll. Das Höhenkoten Tool bleibt als "veraltet" markiert und ist bitte nicht mehr zu verwenden.

Mein Zugang ist (falls es wirklich umgesetzt werden soll), dass man im XYZ Tool optional noch auswählen kann, was in der Karte beschriftet werden soll: Koordinaten oder eine Höhenkoten. Also unter der ``GK M34`` Auswahlliste (siehe Bild oben) noch eine mit:

* nichts beschriften
* Koordinaten beschriften
* DGM 10m beschriften
* ...

Leider ist das Problem mit dem Beschriften nicht ganz trivial. Im WebGIS 4 wurde alles *serverseitig* gerendert und nur ein (Karten) Bild mit alles Marker und Texten ausgeliefert. Jetzt müssen die Texte irgendwie *clientseitig* zu den Markern dazu gerendert werden. Die Frage ist, ob das technisch überhaupt möglich ist. Ich werde das einmal prüfen.
